dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/WebContainer.java

```diff
@@ -2004,4 +2016,31 @@ private static class ReadCipherBitSize {
 
     }
     // ================== CLASS ==================
+    
+    private static class VHostCacheKey {
+        private final String serverName;
+        private final int serverPort;
+
+        VHostCacheKey (final String name, final int port) {
+            this.serverName = name;
+            this.serverPort = port;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if(this == o)
+                return true;
+            if(o == null)
+                return false;
+            if (o.getClass() != VHostCacheKey.class)
+                return false;
+            VHostCacheKey that = (VHostCacheKey) o;
+            return this.serverPort == that.serverPort && this.serverName.equals(that.serverName);
```

Comment:
```suggestion
            return this.serverPort == that.serverPort && (this.serverName == null ? that.serverName == null : this.serverName.equals(that.serverName));
```
getHostAliasKey indicates that the serverName can be null.